src/main/java/org/alfresco/messaging/camel/routes/OnContentUpdateRenditionRoute.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2018 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.messaging.camel.routes;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.camel.spring.SpringRouteBuilder;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Rendition listener for on content update raw event.
+ * 
+ * @author Cristian Turlica
+ */
+@Component
+public class OnContentUpdateRenditionRoute extends SpringRouteBuilder
+{
```

Comment:
This route is built on top of the messaging subsystem, this is in contrast with how the transform event consumer route (REPO-3692) is configured (separately from the messaging subsystem, although using the same camel context). I think we should define a common mechanism for adding these routes. This will depend on whether we want to keep and build on top of the messaging subsystem, in which case we should extend the routs package scan to enterprise packages and configure the REPO-3692 route that way. If not, we should probably configure this route differently. What do you think?

Reply:
Or maybe this can be done as part of REPO-3811..